packages/config/src/layer2s/degate.ts

```diff
@@ -0,0 +1,265 @@
+import { assert, ProjectId, UnixTime } from '@l2beat/shared'
+import { utils } from 'ethers'
+
+import { ProjectDiscovery } from '../discovery/ProjectDiscovery'
+import { formatSeconds } from '../utils/formatSeconds'
+import {
+  DATA_AVAILABILITY,
+  EXITS,
+  FORCE_TRANSACTIONS,
+  makeBridgeCompatible,
+  NEW_CRYPTOGRAPHY,
+  OPERATOR,
+  RISK_VIEW,
+  STATE_CORRECTNESS,
+} from './common'
+import { Layer2 } from './types'
+
+const discovery = new ProjectDiscovery('degate')
+
+const forcedWithdrawalDelay = formatSeconds(
+  discovery.getContractValue<number[]>('ExchangeV3', 'getConstants')[2],
+)
+const maxAgeDepositUntilWithdrawable = formatSeconds(
+  discovery.getContractValue<number>(
+    'ExchangeV3',
+    'getMaxAgeDepositUntilWithdrawable',
+  ),
+)
+
+const forcedWithdrawalFee = discovery.getContractValue<number>(
+  'LoopringV3',
+  'forcedWithdrawalFee',
+)
+
+const forcedWithdrawalFeeString = `${utils.formatEther(
+  forcedWithdrawalFee,
+)} ETH`
+
+export const degate: Layer2 = {
+  type: 'layer2',
+  id: ProjectId('degate'),
+  display: {
+    name: 'DeGate',
+    slug: 'degate',
+    description:
+      'DeGate is an app-specific ZK rollup that enables a Trustless, fast and low-fee decentralized order book exchange, helping users to trade easy and sleep easy.',
+    purpose: 'Exchange',
+    links: {
+      websites: ['https://degate.com/'],
+      apps: ['https://app.degate.com/'],
+      documentation: ['https://docs.degate.com/'],
+      explorers: [],
+      repositories: ['https://github.com/degatedev/protocols'],
+      socialMedia: [
+        'https://twitter.com/DeGateDex',
+        'https://discord.gg/RFVDKGemJb',
+        'https://youtube.com/@degatedex1718',
+        'https://medium.com/degate',
+        'https://mirror.xyz/0x078a601f492043C8e7D0E15B0F8815f58b4c342f',
+      ],
+    },
+  },
+  config: {
+    associatedTokens: ['DG'],
+    escrows: [
+      discovery.getEscrowDetails({
```

Comment:
escrows should be hardcoded, see here: https://www.notion.so/l2beat/Config-progress-ec546917540243fbb0ec599a7af646f3